### PR TITLE
6135 backend

### PR DESCRIPTION
**Pull request by ggainey**
Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/app/importexport.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    Content,
+    ContentArtifact,
+)
+from pulpcore.app.models.repository import (
+    Repository,
+    RepositoryVersion,
+)
+
+
+class RepositoryResource(resources.ModelResource):
```
Comment (daviddavis) [inline pulpcore/app/importexport.py:15]: These seem to duplicate what's in `modelresource.py`?

File: pulpcore/app/viewsets/exporter.py [hunk 2]
```diff
@@ -71,3 +81,46 @@ class PulpExporterViewSet(ExporterViewSet):
     endpoint_name = 'pulp'
     serializer_class = PulpExporterSerializer
     queryset = PulpExporter.objects.all()
+
+
+class ExportViewSet(NamedModelViewSet,
```
Comment (daviddavis) [inline pulpcore/app/viewsets/exporter.py:86]: Looks like this is defined twice?

File: pulpcore/app/viewsets/exporter.py [hunk 3]
```diff
@@ -71,3 +81,46 @@ class PulpExporterViewSet(ExporterViewSet):
...
+
+    def create(self, request, exporter_pk):
+        """
+        Export a set of repository-versions.
+
+        """
+        exporter = PulpExporter.objects.get(pk=exporter_pk).cast()
+
+        result = enqueue_with_reservation(
+            exporter.pulp_export,
```
Comment (daviddavis) [inline pulpcore/app/viewsets/exporter.py:122]: I know I created this pattern for file system exporter, but I think the code should probably just live in `/app/tasks/export.py`. And I would probably just drop this `pulp_export()` method.

File: pulpcore/app/importexport.py (hunk 1 above)
Comment (ggainey) [reply pulpcore/app/importexport.py:15]: Indeed - leftover froma  refactoring

File: pulpcore/app/viewsets/exporter.py (hunk 2 above)
Comment (ggainey) [reply pulpcore/app/viewsets/exporter.py:86]: Yeah, no idea how that happened

File: pulpcore/app/viewsets/exporter.py (hunk 3 above)
Comment (ggainey) [reply pulpcore/app/viewsets/exporter.py]: Refactored a number of things and repushed.

File: pulpcore/app/tasks/export.py [hunk 4]
```diff
@@ -74,18 +77,38 @@ def pulp_export(pulp_exporter):
...
     for repo in repositories:
         version = repo.latest_version()
-        export.export_repository_version(version)
+        # Check version-content to make sure we're not being asked to export an on_demand repo
+        content_artifacts = ContentArtifact.objects.filter(content__in=version.content)
+        if content_artifacts.filter(artifact=None).exists():
+            RuntimeError(_("Remote artifacts cannot be exported."))
+
+        repo_versions.append(version)
+        for ca in content_artifacts:
```
Comment (ggainey) [inline pulpcore/app/tasks/export.py:102]: This is a horribly-inefficient way to get all the Artifacts for a repo-version - what's the Correct Way?

**Pull request merged**